oam/oamcpp/oamcache.cpp

```diff
@@ -237,4 +233,79 @@ string OamCache::getModuleName()
   return moduleName;
 }
 
+bool OamCache::isAccessibleBy(int dbRoot, int pmId)
+{
+  return (*dbRootPMMap)[dbRoot].count(pmId);
```

Comment:
(and below) Is it ok that elements can be implicitly created in the map if `dbRoot` is not there?

Reply:
@mariadb-SergeyZefirov ^

Reply:
It is OK.

First, this is refactored from old code that used the same logic, possibly inserting element if there was none.

Second, even if dbRoot is inserted, the nodes it is associated with is empty. It is unaccessible anyway.

Third, memory overhead is negligible.

Reply:
Now we have a single point of responsibility, so we can easily add verification that the requested dbroot exists. IMO, it's better to end the operation as early as possible with an error than to try to figure out where incorrect entries in the map came from.

I do not insist on making these changes as part of this pull request, but in general it would be good to have them.